nx/lib/nx/tensor.ex

```diff
@@ -143,8 +143,62 @@ defmodule Nx.Tensor do
   end
 
   @impl true
-  def get_and_update(_tensor, _index, _update) do
-    raise "Access.get_and_update/3 is not yet supported by Nx.Tensor"
+  def get_and_update(%Nx.Tensor{shape: {}} = tensor, _index, _update) do
+    raise ArgumentError,
+          "cannot use the tensor[index] syntax on scalar tensor #{inspect(tensor)}"
+  end
+
+  def get_and_update(tensor, index, update) do
+    if index < 0 or index >= elem(tensor.shape, 0) do
+      raise ArgumentError,
+            "index #{index} is out of bounds for axis 0 in shape #{inspect(tensor.shape)}"
+    end
+
+    {:ok, current_value} = fetch(tensor, index)
+    {_, new_value} = update.(current_value)
+
+    if current_value.shape != new_value.shape do
+      raise ArgumentError,
+            "the shape #{inspect(new_value.shape)} of the new value does not match the current value shape #{inspect(current_value.shape)}"
+    end
+
+    {before_indices, after_indices} = get_indices(index, tensor.shape)
+    {before_lengths, after_lengths} = get_lengths(index, tensor.shape)
+
+    slice_before = get_slice(tensor, before_indices, before_lengths)
+    slice_after = get_slice(tensor, after_indices, after_lengths)
```

Comment:
do we really need to get slices?

There is Nx.indexed_put and Nx.put_slice, one of which I would expect to be used here instead of getting slices + concatenation

Reply:
Yes! That is a very good catch! I just re-read the issue and see it was already mentioned there, I misread before and thought it said slices. I just tried it with `put_slices` and it simplifies the code a lot.